Run golanglint-ci when a golanglint-ci config file exists

freeformz opened the pull request:
Many teams generally include some sort of linting pass in their CI configs. This is basically one tool to rule them all with a nice sentinel config file that can be used to detect intent.

cc @meekmichael

Commit:
Initial golanglint-ci support

When a golanglint-ci config file exists, run golanglint-ci

Commit:
Update testpack details

This also adds `make testpack` to run a single FIXTURE as a testpack.

Commit:
Exit loading common.sh if it's already loaded

Commit:
Add golanglint-ci support

When one of .golangci.{yml,toml,json} exists run golanglint-ci.
Includes a fixture with a test in it as well.

Commit:
Sort the assets needed for testing.

jmorrell commented on bin/test:
```diff
@@ -45,8 +45,12 @@ output=$(mktemp)
 case "${TOOL}" in
     gomodules)
         cd "${build}"
+        if [ -f "${build}/.golangci.yml" -o -f "${build}/.golangci.toml" -o -f "${build}/.golangci.json" ]; then
```
Could you add an `mcount` to track usage here?

freeformz replied:
Sure. But, what specifically do you want to track:
* The fact that there is a config file?
* That there isn't a config file?
* That the config file is yaml? Or toml? Or json?
* That golangci-lint was run? Or not run?
* All of the above?
* Something else?

jmorrell replied:
specifically that golangci-lint was run and this feature was used

freeformz replied:
@jmorrell Reminder: this is testpack code. Does the service that execute testpack stuff support the metrics stuff that the build system supports? Last time I checked, which was admittedly a long time ago, it didn't.

jmorrell replied:
ahh, that's a very good point. Could you add a check for these files to `bin/compile` then?

jmorrell commented on lib/common.sh:
```diff
@@ -4,6 +4,9 @@
 # allow apps to specify cgo flags. The literal text '${build_dir}' is substituted for the build directory
 
+test -n "${_COMMON_LOADED:-}" && exit 0
+export _COMMON_LOADED="true"
```
This is a nifty bash pattern

Commit:
Fix/explain GOPATH logic

When modules are being used:
* $GOPATH/pkg/mod is used as the module cache
* test-compile/compile will use a GOPATH of ${build}/.heroku/go-path

So re-use the same GOPATH now to use the cached modules.

Commit:
Standardize on = as per the man page.

== Also works, but is there for "compatibility".

Commit:
Small refinements

Test helper setup threw away the capture of test-compile. Use
`continue_capture` if you need to capture the output of > 1 command.

Commit:
Adjust common bits so things work
